@Test public void requestWhenCustomJwtValidatorFailsThenCorrespondingErrorMessage() throws Exception {
  this.spring.register(WebServerConfig.class,CustomJwtValidatorConfig.class).autowire();
  this.authz.enqueue(this.jwks("Default"));
  String token=this.token("ValidNoScopes");
  OAuth2TokenValidator<Jwt> jwtValidator=this.spring.getContext().getBean(CustomJwtValidatorConfig.class).getJwtValidator();
  OAuth2Error error=new OAuth2Error("custom-error","custom-description","custom-uri");
  when(jwtValidator.validate(any(Jwt.class))).thenReturn(OAuth2TokenValidatorResult.failure(error));
  this.mvc.perform(get("/").with(bearerToken(token))).andExpect(status().isUnauthorized()).andExpect(header().string(HttpHeaders.WWW_AUTHENTICATE,containsString("custom-description")));
}
